libraries/cfstore/TESTS/testgroup/example1/example1.cpp

```diff
@@ -0,0 +1,850 @@
+            ret = cfstore_drv->Delete(ctx->callback_handle);
+            CFSTORE_EX1_TEST_ASSERT_MSG(ret >= ARM_DRIVER_OK, "%s:Error: Close() failed (ret=%ld)\r\n", __func__, ret);
+            if(ret >= ARM_DRIVER_OK && ctx->caps.asynchronous_ops == false) {
+                ctx->state = CFSTORE_EX_STATE_DELETE_DONE;
+                break;
+            } else if(ret >= ARM_DRIVER_OK && ctx->caps.asynchronous_ops == true) {
+                // await pending notification of completion.
+                break;
+            }
+            CFSTORE_EX1_TEST_ASSERT_MSG(false, "%s:Error: unknown error (line=%u)\r\n", __func__, __LINE__);
+            break;
+
+        case CFSTORE_EX_STATE_DELETE_DONE:
+            CFSTORE_EX1_LOG("DELETE_DONE%s", "\r\n");
+            CFSTORE_EX1_TEST_ASSERT_MSG(ctx->callback_status >= ARM_DRIVER_OK, "%s:Error: Delete() completion failed (status=%ld)\r\n", __func__, ctx->callback_status);
+            CFSTORE_EX1_TEST_ASSERT_MSG(ctx->callback_handle == NULL, "%s:Error: the cfstore_ex_callback(cmd_code==CFSTORE_OPCODE_DELETE) received non-NULL handle (%p)\r\n", __func__, ctx->callback_handle);
+            CFSTORE_HANDLE_SWAP(ctx->hkey_prev, ctx->hkey_next);
+            ctx->state = CFSTORE_EX_STATE_FINDING2;
+            // intentional fall-through
+
+        case CFSTORE_EX_STATE_FINDING2:
+            CFSTORE_EX1_LOG("FINDING2%s", "\r\n");
+            // note that cfstore_ex_callback() for cmd_code==CFSTORE_OPCODE_FIND can be invoked before Find() has returned
+            ret = cfstore_drv->Find("*", ctx->hkey_next, ctx->hkey_prev);
+            CFSTORE_EX1_TEST_ASSERT_MSG(ret == ARM_CFSTORE_DRIVER_ERROR_KEY_NOT_FOUND, "%s:Error: Find() failed to return expected value of ARM_CFSTORE_DRIVER_ERROR_KEY_NOT_FOUND (ret=%ld)\r\n", __func__, ret);
+            if(ret == ARM_CFSTORE_DRIVER_ERROR_KEY_NOT_FOUND && ctx->caps.asynchronous_ops == false) {
+                ctx->state = CFSTORE_EX_STATE_FIND_DONE2;
+                break;
+            } else if(ret == ARM_CFSTORE_DRIVER_ERROR_KEY_NOT_FOUND && ctx->caps.asynchronous_ops == true) {
+                // await pending notification of completion.
+                break;
+            }
+            CFSTORE_EX1_TEST_ASSERT_MSG(false, "%s:Error: unknown error (line=%u)\r\n", __func__, __LINE__);
+            break;
+
+        case CFSTORE_EX_STATE_FIND_DONE2:
+            CFSTORE_EX1_LOG("FIND_DONE2%s", "\r\n");
+            CFSTORE_EX1_TEST_ASSERT_MSG(ctx->callback_status == ARM_CFSTORE_DRIVER_ERROR_KEY_NOT_FOUND, "%s:Error: Find() completion should have been ARM_CFSTORE_DRIVER_ERROR_KEY_NOT_FOUND (status=%ld)\r\n", __func__, ctx->callback_status);
+        // todo: delete next 2 lines when fix flash journal issue mentioned below
+            ctx->state = CFSTORE_EX_STATE_UNINITIALIZING;
+            // intentional fall-through
+
+        /* todo: restore this code when issue with flushing empty flash to flash journal has been fixed
```

Comment:
Same question for this block of commented code, although it looks like there's a plan on restoring this code based on the above `todo`